### ISSUE TITLE
Ability to handle statement / resultSet getWarnings()

### ISSUE THREAD

**Issue by jukkasi** [EvidentSolutions/dalesbred#43]
Because we are using MariaDB which sometimes does not behave that nice but only produces warnings when it actually should throw errors, I'm looking for similar feature that (Spring) JdbcTemplate has, which is flag ignoreWarnings = false that throws exceptions if there are warnings in result.

I understand that this can be achieved by using ResultSet.getWarnings() but because we are not currently using ResultSet in any way but only under the hood (= only Dalesbred built-in row mappers), I'm looking for more generic solution.

Also preparedStatement.getWarnings() cannot be used because stmt is well encapsulated inside Database class implementation. Using custom transaction manager I am able to access Connection after commit but calling getWarnings() for Connection is a different case and not sufficient.

Any ideas?

**Comment by komu** [comment]
Sounds like a good idea. Currently there's no way to customize this, but this is the only place in Dalesbred's code that deals with ResultSets so it should be a matter of implementing it here:

https://github.com/EvidentSolutions/dalesbred/blob/eaf01af193e15b0a9e84196b161d3126fb3a093d/dalesbred/src/main/java/org/dalesbred/Database.java#L279-L282

However, I'm torn between a specific high-level flag like `ignoreWarnings = false` and a more general event-listener that could be registered to `Database` and would be called every time that query has been executed (among various other events). Something like:

```java
 try (ResultSet resultSet = ps.executeQuery()) { 
     long elapsedMillis = currentTimeMillis() - startTime;
     logQueryExecution(query, elapsedMillis); 
     callback.queryExecuted(query, resultSet, elapsedMillis);
     return processor.process(resultSet); 
} 
```

I'll have to give some though into this.

**Comment by jukkasi** [comment]
Yes, some kind of hook / event calback / interceptor mechanism could work and I believe something similar was requested in https://github.com/EvidentSolutions/dalesbred/issues/23

However I'm not sure if resultSet or preparedStatement should be accessible, or perhaps both.
